File: gome-android-client/src/main/java/com/guardanis/gome/commands/KeycodeCommand.java
package com.guardanis.gome.commands;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

public class KeycodeCommand implements Command {

    public static final int VK_INSERT = 155;
    public static final int VK_HOME = 36;
    public static final int VK_END = 35;
    public static final int VK_ESCAPE = 27;

    private int value;
    private List<String> wrappedValues;

    public KeycodeCommand(int value, List<String> wrappedValues) {
        this.value = value;
        this.wrappedValues = wrappedValues;
    }

    @Override
    public String getActionIdentifier() {
        return "key";
    }

    @Override
    public JSONObject toJson() throws JSONException {
        JSONObject json = new JSONObject()
                .put("type", "action")
                .put("value", value);

        if (wrappedValues != null) {
            JSONArray wrapped = new JSONArray();

            for (String s : wrappedValues)
                wrapped.put(s);

            json.put("wrapped", wrapped);
        }

        return json;
    }
}

File: gome-swift-client/gome/KeycodeCommand.swift
import UIKit

class KeycodeCommand: Command {
    
    public static let VK_INSERT = 155
    public static let VK_HOME = 36
    public static let VK_END = 35
    public static let VK_ESCAPE = 27
    
    private var value: Int!
    private var wrappedValues: [String]?
    
    init(_ value: Int){
        self.value = value
    }
    
    init(_ value: Int, wrappedValues: [String]?){
        self.value = value
        self.wrappedValues = wrappedValues
    }
    
    func getActionId() -> String {
        return "key"
    }
    
    func getData() -> [String: Any] {
        var data = [String: Any]()
        data["type"] = "action"
        data["value"] = value
        
        if let wrapped = wrappedValues {
            var wrappedData = [String]()
            
            for item in wrapped {
                wrappedData.append(item)
            }
            
            data["wrapped"] = wrappedData
        }
        
        return data
    }
    
}
